raft/src/main/java/org/apache/kafka/raft/internals/BatchAccumulator.java

```diff
@@ -284,6 +284,8 @@ private int validateMemoryRecordAndReturnCount(MemoryRecords memoryRecord) {
             );
         } else if (numberOfRecords == null) {
             throw new IllegalArgumentException("valueCreator didn't create a batch with the count");
+        } else if (numberOfRecords < 1) {
```

Comment:
Is it possible to add new UT to `BatchAccumulatorTest` for this case?

Reply:
Added the test.